core/unix/signal.c

```diff
@@ -6172,7 +6205,7 @@ terminate_via_kill(dcontext_t *dcontext)
      * this may not exit all threads in the address space
      */
     block_cleanup_and_terminate(
-        dcontext, SYS_kill,
+        dcontext, IF_MACOS(SYSCALL_NUM_MARKER_UNIX +) SYS_kill,
```

Comment:
There are multiple calls to *cleanup_and_terminate.  How are the others working on 64-bit?  Here just one is being changed.  Better to adjust the syscall number inside cleanup_and_terminate (and assume it's BSD and not Mach or Machdep) so all calls work and we continue using SYS_ constants.

Reply:
Before, I wasn't sure if we want to assume the syscall arg to cleanup_and_terminate to be BSD always. I agree that would be better. Done.

Reply:
Xref some mach ones in x86.asm -- one is dynamorio_sys_exit which might be the only concern if someone passed that mach number.

Reply:
The usages in x86.asm seem fine, as the syscall is decided statically, so we know whether they are bsd/mach.

```
        mov      eax, MACH_thread_self_trap
# ifdef X64
        or       eax, SYSCALL_NUM_MARKER_MACH
# else
```

```
        mov      eax, SYS_bsdthread_terminate
        or       eax, SYSCALL_NUM_MARKER_BSD
```

```
        mov      eax, MACH_semaphore_signal_all_trap
        or       eax, SYSCALL_NUM_MARKER_MACH
```

dynamorio_syscall in `core/drlibc/drlibc_x86.asm` blindly assumes BSD though.